api/controllers/webhook.js

```diff
@@ -1,148 +1,22 @@
-const crypto = require('crypto');
-const Sequelize = require('sequelize');
-const config = require('../../config');
-const GithubBuildHelper = require('../services/GithubBuildHelper');
-const EventCreator = require('../services/EventCreator');
-const {
-  Build, User, Site, Event,
-} = require('../models');
+const { wrapHandlers } = require('../utils');
+const Webhooks = require('../services/Webhooks');
 
-const signBlob = (key, blob) => `sha1=${crypto.createHmac('sha1', key).update(blob).digest('hex')}`;
-
-const findSiteForWebhookRequest = (request) => {
-  const owner = request.body.repository.full_name.split('/')[0].toLowerCase();
-  const repository = request.body.repository.full_name.split('/')[1].toLowerCase();
-
-  return Site.findOne({
-    where: {
-      owner,
-      repository,
-      buildStatus: { [Sequelize.Op.ne]: 'inactive' },
-    },
-    include: [{ model: User }],
-  })
-    .then((site) => {
-      if (!site) {
-        throw new Error(`Unable to find an active Federalist site with ${owner}/${repository}`);
-      } else {
-        return site;
-      }
-    });
-};
-
-const organizationWebhookRequest = async (payload) => {
-  const {
-    action, membership, organization,
-  } = payload;
-  const { login: orgName } = organization;
-
-  if (orgName !== config.federalistUsers.orgName) {
-    return;
-  }
-
-  const { login } = membership.user;
-  const username = login.toLowerCase();
-  let user = await User.findOne({ where: { username } });
-
-  if (['member_added', 'member_removed'].includes(action)) {
-    const isActive = action === 'member_added';
-
-    if (!user && isActive) {
-      user = await User.create({ username });
-    }
-    if (user) {
-      if (isActive !== user.isActive) {
-        await user.update({ isActive });
-      }
-      EventCreator.audit(Event.labels.FEDERALIST_USERS_MEMBERSHIP, user, payload);
-    }
-  }
-};
-
-const signWebhookRequest = (request) => {
-  const webhookSecret = config.webhook.secret;
-  const requestBody = JSON.stringify(request.body);
-
-  const signature = request.headers['x-hub-signature'];
-  const signedRequestBody = signBlob(webhookSecret, requestBody);
-
-  if (!signature) {
-    throw new Error('No X-Hub-Signature found on request');
-  } else if (signature !== signedRequestBody) {
-    throw new Error('X-Hub-Signature does not match blob signature');
-  }
-};
-
-const createBuildForWebhookRequest = async (request) => {
-  const { login } = request.body.sender;
-  const { pushed_at: pushedAt } = request.body.repository;
-  const username = login.toLowerCase();
-  const site = await findSiteForWebhookRequest(request);
-
-  let user = site.Users.find(u => u.username === username);
-  if (!user) {
-    user = await User.findOne({ where: { username } });
-    if (user) {
-      await site.addUser(user);
-    }
-  }
-  if (user) {
-    await user.update({ pushedAt: new Date(pushedAt * 1000) });
-  }
+module.exports = {
+  ...wrapHandlers({
```

Comment:
Spreading the `wrapHandlers` here but the other controllers you're not.  Just wondering about convention and consistency.

Reply:
it's spread because I was exporting extra functions, but then i removed them, so I will clean this up.

Reply:
Copy that